buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -328,7 +328,7 @@ static void getBytes(AbstractByteBuf buf, long addr, int index, ByteBuffer dst)
     static void setBytes(AbstractByteBuf buf, long addr, int index, ByteBuf src, int srcIndex, int length) {
         buf.checkIndex(index, length);
         checkNotNull(src, "src");
-        if (AbstractByteBuf.isInvalid(srcIndex, length, src.capacity())) {
+        if (isOutOfBounds(srcIndex, length, src.capacity())) {
             throw new IndexOutOfBoundsException("srcIndex: " + srcIndex);
```

Comment:
Also I would prefer if you revert all of these changes above as these not belong to the header change if you ask me.